storage/oasis/nodeapi/history/history.go

```diff
@@ -0,0 +1,177 @@
+package history
+
+import (
+	"context"
+
+	beacon "github.com/oasisprotocol/oasis-core/go/beacon/api"
+	"github.com/oasisprotocol/oasis-core/go/common"
+	consensus "github.com/oasisprotocol/oasis-core/go/consensus/api"
+	genesis "github.com/oasisprotocol/oasis-core/go/genesis/api"
+	governance "github.com/oasisprotocol/oasis-core/go/governance/api"
+	registry "github.com/oasisprotocol/oasis-core/go/registry/api"
+	roothash "github.com/oasisprotocol/oasis-core/go/roothash/api"
+	scheduler "github.com/oasisprotocol/oasis-core/go/scheduler/api"
+	staking "github.com/oasisprotocol/oasis-core/go/staking/api"
+
+	"github.com/oasisprotocol/oasis-indexer/storage/oasis/nodeapi"
+	"github.com/oasisprotocol/oasis-indexer/storage/oasis/nodeapi/cobalt"
+	"github.com/oasisprotocol/oasis-indexer/storage/oasis/nodeapi/damask"
+)
+
+var _ nodeapi.ConsensusApiLite = (*HistoryConsensusApiLite)(nil)
+
+func NotImplemented() nodeapi.ConsensusApiLite {
+	panic("not implemented")
+}
+
+type Record struct {
+	ArchiveName   string
+	ConstructAPI  func() nodeapi.ConsensusApiLite
+	GenesisHeight int64
+}
+
+type History struct {
+	Records []*Record
+}
+
+var Mainnet = History{
+	Records: []*Record{
+		{
+			// https://github.com/oasisprotocol/mainnet-artifacts/releases/tag/2022-04-11
+			ArchiveName:   "damask",
+			ConstructAPI:  damask.NewDamaskConsensusApiLite,
+			GenesisHeight: 8048956,
```

Comment:
I think we'll also need the lowest round of each supported runtime. That way, we'll know which node (URL) to hit for each round.

Reply:
Agreed. I should have been clearer; I intended my comment in the context of this single-node config.
For each runtime, we want the lowest round for which the data is available _on this node version_ (cobalt, damask).

In case you meant something even more fine-grained by "node dependent": It's technically true that each individual node deployment could have differently pruned history, but that's also true in consensus. We should assume unpruned history, and then make sure at the ops level that that's how our nodes are deployed.

Reply:
what's the limit in that case?

Reply:
For consensus, it's the height at which the dump-restore happened, i.e. GenesisHeight.
For runtimes, the first round that was finalized at or after that height. ... oh ... do you reckon a Damask node can potentially have all of Emerald history including Cobalt, because the runtime's statetree format hasn't changed? I haven't tested but I think that's not the case.

Reply:
I think we can get the round number from the genesis roothash entries

Reply:
continuing in clickup https://app.clickup.com/t/866a0fpr3